### DISCUSSION TITLE
Accessing actions from LSP

### DISCUSSION THREAD

**Discussion by LucasPickering** [zed-industries/zed#16434] [Feature Requests]
Rust-analyzer provides helpful actions (not sure if that's correct term) that in VSCode can be bound to keybindings. Two examples I'd really like to use in Zed are:
- [Open Docs](https://rust-analyzer.github.io/manual.html#open-docs)
- [Move Item](https://rust-analyzer.github.io/manual.html#move-item)

I've done a lot of googling and reading but can't find any references to how to bind these in Zed, or otherwise access them. Is it currently possible? I read through the [docs on LSPs](https://zed.dev/docs/configuring-languages#using-language-server-features) but didn't see anything in there.

**Comment by SomeoneToIgnore** [comment]
Both of the commands come from https://github.com/rust-lang/rust-analyzer/blob/master/editors/code/src/lsp_ext.ts
Zed does not support anything but exactly 1 extra, non-standard LSP command: https://github.com/zed-industries/zed/blob/6f93b42ecb239e9c494a1f969a183b30787833a3/crates/project/src/lsp_ext_command.rs#L14

So there's nothing to bind to as there's no such support in Zed at all.

**Reply by LucasPickering** [reply]
That's what I suspected. Do you know if there's a tracking issue for this yet? I wasn't able to find one

**Reply by SomeoneToIgnore** [reply]
There's none, and realistically, it might be faster to create a PR than an issue.

**Reply by estk** [reply]
There is this issue: https://github.com/zed-industries/zed/issues/13104

Its surprising to me that the core team hasnt needed this, can anyone from the core team comment?
